services/scan/src/precinct_scanner_app.ts

```diff
@@ -0,0 +1,454 @@
+import { OkResponse, Scan } from '@votingworks/api';
+import * as streams from 'memory-streams';
+import { Buffer } from 'buffer';
+import * as fsExtra from 'fs-extra';
+import {
+  BallotPageLayoutSchema,
+  BallotPageLayoutWithImage,
+  safeParse,
+  safeParseElectionDefinition,
+  safeParseJson,
+} from '@votingworks/types';
+import { assert, find } from '@votingworks/utils';
+import express, { Application } from 'express';
+import { readFile } from 'fs-extra';
+import multer from 'multer';
+import { z } from 'zod';
+import { PrecinctScannerStateMachine } from './plustek_machine';
+import { pdfToImages } from './util/pdf_to_images';
+import { Workspace } from './util/workspace';
+import { createInterpreter, loadLayouts } from './simple_interpreter';
+import { backup } from './backup';
+
+function sum(nums: number[]) {
+  return nums.reduce((a, b) => a + b, 0);
+}
+
+type NoParams = never;
+
+async function configureMachine(
+  machine: PrecinctScannerStateMachine,
+  workspace: Workspace
+) {
+  const { store } = workspace;
+  const electionDefinition = store.getElectionDefinition();
+  assert(electionDefinition);
+  const layouts = await loadLayouts(store);
+  assert(layouts);
+  const interpreter = createInterpreter({
+    electionDefinition,
+    ballotImagesPath: workspace.ballotImagesPath,
+    testMode: store.getTestMode(),
+    layouts,
+  });
+
+  machine.configure(store, interpreter);
+}
+
+export function buildPrecinctScannerApp(
```

Comment:
@eventualbuddha probably my biggest question is how to organize the shared bits of the scan service that both central scanner and precinct scanner use. To start, I just copy/pasted all the routes used by the precinct scanner and tweaked them to not use `Importer` (instead inlining the relevant pieces of `Importer`). Should I leave things this way? Seems like a lot of duplicated code. Part of me thinks these could eventually be two separate services, in which case splitting them seems ok. But then again there's a lot of shared logic (though I guess most of it is just glue code/request validation). Curious to hear your thoughts

Reply:
I think a reasonable compromise for now would be to have them in the same service but distinct `Application`s. That way the shared code is relatively easy to manage and we just start the application that makes sense. Longer term perhaps we do make them distinct services, but that'll require an investment in factoring out the shared code like `Store` etc.

Reply:
Ok that sounds good. I'll continue down this path then.